src/app/clusters/microwave-oven-control-server/microwave-oven-control-server.cpp

```diff
@@ -90,43 +119,141 @@ void Instance::SetCookTime(uint32_t cookTime)
+        if (HasFeature(MicrowaveOvenControl::Feature::kPowerAsNumber))
+        {
+            if (HasFeature(MicrowaveOvenControl::Feature::kPowerNumberLimits))
+            {
+                ReturnErrorOnFailure(aEncoder.Encode(mDelegate->GetMaxPower()));
+            }
+            else
+            {
+                ReturnErrorOnFailure(aEncoder.Encode(kDefaultMaxPower));
+            }
+        }
+        else
+        {
+            ChipLogError(Zcl, "Microwave Oven Control: feature is not supported");
+            return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+        }
     }
     break;
 
     case MicrowaveOvenControl::Attributes::PowerStep::Id: {
-        ReturnErrorOnFailure(aEncoder.Encode(mDelegate->GetPowerStep()));
+        if (HasFeature(MicrowaveOvenControl::Feature::kPowerAsNumber))
+        {
+            if (HasFeature(MicrowaveOvenControl::Feature::kPowerNumberLimits))
+            {
+                ReturnErrorOnFailure(aEncoder.Encode(mDelegate->GetPowerStep()));
+            }
+            else
+            {
+                ReturnErrorOnFailure(aEncoder.Encode(kDefaultPowerStep));
+            }
+        }
+        else
+        {
+            ChipLogError(Zcl, "Microwave Oven Control: feature is not supported");
+            return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+        }
+    }
+    break;
+
+    case MicrowaveOvenControl::Attributes::SupportedWatts::Id: {
+        if (HasFeature(MicrowaveOvenControl::Feature::kPowerInWatts))
+        {
+            aEncoder.EncodeList([delegate = mDelegate](const auto & encoder) -> CHIP_ERROR {
+                uint16_t wattRating;
+                uint8_t index  = 0;
+                CHIP_ERROR err = CHIP_NO_ERROR;
+                while ((err = delegate->GetWattSettingByIndex(index, wattRating)) == CHIP_NO_ERROR)
+                {
+                    ReturnErrorOnFailure(encoder.Encode(wattRating));
+                    index++;
+                }
+                if (err == CHIP_ERROR_NOT_FOUND)
+                {
+                    return CHIP_NO_ERROR;
+                }
+                return err;
+            });
+        }
+        else
+        {
+            ChipLogError(Zcl, "Microwave Oven Control: feature is not supported");
+            return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+        }
+    }
+    break;
+
+    case MicrowaveOvenControl::Attributes::SelectedWattIndex::Id: {
+        if (HasFeature(MicrowaveOvenControl::Feature::kPowerInWatts))
+        {
+            ReturnErrorOnFailure(aEncoder.Encode(mDelegate->GetCurrentWattIndex()));
+        }
+        else
+        {
+            ChipLogError(Zcl, "Microwave Oven Control: feature is not supported");
+            return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+        }
+    }
+    break;
+
+    case MicrowaveOvenControl::Attributes::WattRating::Id: {
+        ReturnErrorOnFailure(aEncoder.Encode(mDelegate->GetCurrentWattRating()));
+    }
```

Comment:
are the extra braces needed? Since we do not declare internal variables they do not seem needed. Comment applies for all case statements here.